core/src/test/java/dk/alexandra/fresco/lib/generic/MaliciousCommitmentComputation.java

```diff
@@ -0,0 +1,53 @@
+package dk.alexandra.fresco.lib.generic;
+
+import dk.alexandra.fresco.commitment.HashBasedCommitment;
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.network.serializers.ByteSerializer;
+import dk.alexandra.fresco.framework.util.Drbg;
+import dk.alexandra.fresco.framework.util.Pair;
+import java.util.List;
+
+public class MaliciousCommitmentComputation extends CommitmentComputation {
+
+  private final ByteSerializer<HashBasedCommitment> commitmentSerializer;
+  private final byte[] value;
+  private final int noOfParties;
+  private final Drbg localDrbg;
+
+  public MaliciousCommitmentComputation(
+      ByteSerializer<HashBasedCommitment> commitmentSerializer,
+      byte[] value, int noOfParties, Drbg localDrbg) {
+    super(commitmentSerializer, value, noOfParties, localDrbg);
+    this.commitmentSerializer = commitmentSerializer;
+    this.value = value;
+    this.noOfParties = noOfParties;
+    this.localDrbg = localDrbg;
+  }
+
+  @Override
+  public DRes<List<byte[]>> buildComputation(ProtocolBuilderNumeric builder) {
+    HashBasedCommitment ownCommitment = new HashBasedCommitment();
+    byte[] ownOpening = ownCommitment.commit(localDrbg, value);
+    return builder.seq(
+        seq -> {
+          if (noOfParties > 2) {
```

Comment:
same as CommitmentComputation.java, would be nice to comment why it's different for <= 2